src/main/java/net/apnic/rdap/rdap/RDAPError.java

```diff
@@ -0,0 +1,98 @@
+package net.apnic.rdap.rdap;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Class represent an error response in the RDAP protocol.
+ *
+ * @see https://tools.ietf.org/html/rfc7483
+ */
+public class RDAPError
+    extends RDAPObject
+{
+    private List<String> description = null;
```

Comment:
I think the need to incrementally build is a result of your design of trying to generalize the factory:

from ErrorController:
```
        RDAPError error =
            rdapObjectFactory.createRDAPObject(RDAPError.class, context)
                .setErrorCode(status.value())
                .setTitle(status.getReasonPhrase());

        if(description != null)
        {
            error.addDescription(description);
        }
```

How about just having a component that supplies the common elements, and then just call the constructor?

```
        RDAPConformance conformance = rdapDefaults.getConformance();
        List<RDAPNotice> notices = rdapDefaults.getNotices(context);

        String errorCode = status.value();
        String title = status.getReasonPhrase();
        List<String> descriptions = Optional.ofNullable(description).map(Collections::singletonList).orElse(Collections.emptyList());
        RDAPError rdapError = new RDAPError(errorCode, title, descriptions, notices, conformance);
```